v2/internal/crdmanagement/manager_test.go

```diff
@@ -260,6 +224,108 @@ func Test_FindNonMatchingCRDs_CRDsWithDifferentConversionsCompareAsEqual(t *test
+				g.Expect(instructions).To(HaveLen(1))
+				g.Expect(instructions[0].FilterResult).To(Equal(crdmanagement.Excluded))
+				g.Expect(instructions[0].FilterReason).To(ContainSubstring("was not matched by CRD pattern and did not already exist in cluster"))
+				apply, _ := instructions[0].ShouldApply()
+				g.Expect(apply).To(BeFalse())
+			},
+		},
+		{
+			name:     "Matches CRDs if pattern matches",
+			goal:     []apiextensions.CustomResourceDefinition{makeBasicCRD("test")},
+			existing: nil,
+			patterns: "testrp.azure.com/*",
+			validate: func(g *WithT, instructions []*crdmanagement.CRDInstallationInstruction) {
+				g.Expect(instructions).To(HaveLen(1))
+				g.Expect(instructions[0].FilterResult).To(Equal(crdmanagement.MatchedPattern))
+				g.Expect(instructions[0].FilterReason).To(Equal("CRD named \"testrp.azure.com/test\" matched pattern \"testrp.azure.com/*\""))
+				g.Expect(instructions[0].DiffResult).To(Equal(crdmanagement.SpecDifferent))
+				apply, _ := instructions[0].ShouldApply()
+				g.Expect(apply).To(BeTrue())
+			},
+		},
+		{
+			name:     "Matches if CRD already installed",
+			goal:     []apiextensions.CustomResourceDefinition{makeBasicCRD("test")},
+			existing: []apiextensions.CustomResourceDefinition{makeBasicCRDWithVersion("test", "1.1")},
+			patterns: "",
+			validate: func(g *WithT, instructions []*crdmanagement.CRDInstallationInstruction) {
+				g.Expect(instructions).To(HaveLen(1))
+				g.Expect(instructions[0].FilterResult).To(Equal(crdmanagement.MatchedExistingCRD))
+				g.Expect(instructions[0].FilterReason).To(Equal("A CRD named \"testrp.azure.com/test\" was already installed, considering that existing CRD for update"))
+				g.Expect(instructions[0].DiffResult).To(Equal(crdmanagement.VersionDifferent))
+				apply, _ := instructions[0].ShouldApply()
+				g.Expect(apply).To(BeTrue())
+			},
+		},
+		{
+			name:     "Matches if CRD already installed and pattern matches, pattern matches wins",
+			goal:     []apiextensions.CustomResourceDefinition{makeBasicCRD("test")},
+			existing: []apiextensions.CustomResourceDefinition{makeBasicCRDWithVersion("test", "1.1")},
+			patterns: "testrp.azure.com/*",
+			validate: func(g *WithT, instructions []*crdmanagement.CRDInstallationInstruction) {
+				g.Expect(instructions).To(HaveLen(1))
+				g.Expect(instructions[0].FilterResult).To(Equal(crdmanagement.MatchedPattern))
+				g.Expect(instructions[0].FilterReason).To(Equal("CRD named \"testrp.azure.com/test\" matched pattern \"testrp.azure.com/*\""))
+				g.Expect(instructions[0].DiffResult).To(Equal(crdmanagement.VersionDifferent))
+				apply, _ := instructions[0].ShouldApply()
+				g.Expect(apply).To(BeTrue())
+			},
+		},
+		{
+			name:     "Pattern matches but CRD already installed, nothing to do",
+			goal:     []apiextensions.CustomResourceDefinition{makeBasicCRD("test")},
+			existing: []apiextensions.CustomResourceDefinition{makeBasicCRD("test")},
+			patterns: "testrp.azure.com/*",
```

Comment:
Consider adding another test with 2 CRDs in a group where `pattern == testrp.azure.com/test` and the other one is ignored

Reply:
Good idea. Added